src/common/ortho.jl

```diff
@@ -7,3 +7,19 @@
         return x[:, 1:size(φk, 2)]
     end
 end
+
+@timing function ortho_lowdin(φk::ArrayType) where {ArrayType <: AbstractArray}
+    S = φk' * φk
+    evals, evecs = eigen(S)
+    # Check for linear dependence: eigenvalues close to zero
+    tol = eps(real(eltype(φk))) * maximum(abs, evals)
+    if any(abs.(evals) .< tol)
+        small_evals = filter(eval -> eval < tol, evals)
+        error("Input vectors are linearly dependent or nearly so 
```

Comment:
This is a low-level routine. Such routines should not `error`. Use an `@assert` or throw an exception. This anyway should not happen.